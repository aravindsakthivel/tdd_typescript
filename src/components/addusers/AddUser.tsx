import { FC, useState, MouseEvent } from 'react';
import { AddUserModal } from './AddUserModal';
import { Button, Row, Col, Typography } from 'antd';
import '../customClassStyle.css';

const { Title } = Typography;

const AddUser: FC = () => {
  const holdClose = false;

  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
  const isModalToggle = (event: MouseEvent<HTMLButtonElement>): void => {
    setIsModalOpen(!isModalOpen);
  };

  return (
    <>
      <Row
        justify="space-between"
        className="general-background-border manage-user-page-class add-user"
      >
        <Col span={5}>
          <Title level={2} style={{ color: '#ffffff' }}>
            Manage User
          </Title>
        </Col>
        <Col span={3}>
          <Button
            key="primary"
            type="text"
            data-testid="add-user-button"
            onClick={isModalToggle}
            size="large"
            className="add-user-button-class button-common-class"
          >
            Add User
          </Button>
        </Col>
      </Row>
      <AddUserModal
        isModalOpen={isModalOpen}
        closeModal={() => setIsModalOpen(false)}
        data-testid="add-user-modal"
        holdClose={holdClose}
      />
    </>
  );
};

export { AddUser };
