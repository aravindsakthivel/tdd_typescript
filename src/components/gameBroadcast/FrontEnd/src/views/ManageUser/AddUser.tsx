import React, { useState, ReactElement } from 'react';
import { Button, Row, Typography } from 'antd';
import AddUserModal from './AddUserModal';
import './addUserStyle.scss';

const { Title } = Typography;

function AddUser(): ReactElement {
  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
  const isModalToggle = (): void => {
    setIsModalOpen(!isModalOpen);
  };

  return (
    <>
      <Row
        justify="space-between"
        className="general-background-border manage-user-page-class add-user">
        <Title level={2} style={{ color: '#ffffff' }}>
          Manage User
        </Title>
        <Button
          key="primary"
          type="text"
          data-testid="add-user-button"
          onClick={isModalToggle}
          size="large"
          className="add-user-button-class button-common-class">
          Add User
        </Button>
      </Row>
      <AddUserModal
        isModalOpen={isModalOpen}
        closeModal={() => setIsModalOpen(false)}
        data-testid="add-user-modal"
      />
    </>
  );
}

export default AddUser;
