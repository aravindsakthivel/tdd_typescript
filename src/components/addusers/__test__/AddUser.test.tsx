import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { mockConfig } from '../../mock';
import { AddUser } from '../AddUser';

afterEach(cleanup);
beforeEach(() => {
  mockConfig();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddUser />, div);
});

it('all the elements presents', () => {
  const { getByTestId } = render(<AddUser />);
  expect(getByTestId('add-user-button')).toHaveTextContent('Add User');
});

it('renders the modal after clicking the add user button', () => {
  const { getByTestId } = render(<AddUser />);
  const adduser = getByTestId('add-user-button');
  fireEvent.click(adduser);
});

it('matches snapshot', () => {
  const tree = renderer.create(<AddUser />).toJSON();
  expect(tree).toMatchSnapshot();
});
