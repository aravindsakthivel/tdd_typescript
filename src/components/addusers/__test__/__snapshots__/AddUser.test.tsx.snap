// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="ant-row ant-row-space-between general-background-border manage-user-page-class add-user"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-5"
      style={Object {}}
    >
      <h2
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": undefined,
            "color": "#ffffff",
          }
        }
      >
        Manage User
      </h2>
    </div>
    <div
      className="ant-col ant-col-3"
      style={Object {}}
    >
      <button
        className="ant-btn ant-btn-text ant-btn-lg add-user-button-class button-common-class"
        data-testid="add-user-button"
        onClick={[Function]}
        type="button"
      >
        <span>
          Add User
        </span>
      </button>
    </div>
  </div>,
  <div />,
]
`;
